manager/controlapi/service.go

```diff
@@ -165,6 +165,10 @@ func validateEndpointSpec(epSpec *api.EndpointSpec) error {
 
 	portSet := make(map[portSpec]struct{})
 	for _, port := range epSpec.Ports {
+		if port.PublishedPort == 0 {
```

Comment:
Does this need a comment? (e.g., if no host port is assigned, ports can never conflict)